internal/rbd/rbd_util.go

```diff
@@ -1570,3 +1570,19 @@ func (ri *rbdImage) isCompatibleEncryption(dst *rbdImage) error {
 
 	return nil
 }
+
+func (ri *rbdImage) isCompatibleThickProvision(dst *rbdVolume) error {
+	thick, err := ri.isThickProvisioned()
+	if err != nil {
+		return err
+	}
+	switch {
+	case thick && !dst.ThickProvision:
+		return fmt.Errorf("cannot create thin volume from thick volume %q", ri)
```

Comment:
The error message mentions creating, but that may not always be correct. use "does not match" or similar instead?

Reply:
keeping error message same as https://github.com/ceph/ceph-csi/pull/2175 any suggestion on the error message?

Reply:
I do not care too much about it :) for now it is only used while creating volumes, so that's fine

Reply:
I think current message is fine: we can also change to  something like this:
`Source ( thin/thick) : Destination ( thin/thick) - Unsupported conversion`

Reply:
Thanks, @humblec as this is a clear error message and fine let's keep it as it is for now. If required we can change it later.